src/axom/primal/operators/winding_number.hpp

```diff
@@ -0,0 +1,391 @@
+// Copyright (c) 2017-2023, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/*!
+ * \file winding_number.hpp
+ *
+ * \brief Consists of methods to compute winding numbers for points 
+ *        with respect to various geometric objects.
+ */
+
+#ifndef AXOM_PRIMAL_WINDING_NUMBER_HPP_
+#define AXOM_PRIMAL_WINDING_NUMBER_HPP_
+
+// Axom includes
+#include "axom/config.hpp"
+
+#include "axom/primal/geometry/Point.hpp"
+#include "axom/primal/geometry/Segment.hpp"
+#include "axom/primal/geometry/Triangle.hpp"
+#include "axom/primal/geometry/Polygon.hpp"
+#include "axom/primal/geometry/Polyhedron.hpp"
+#include "axom/primal/geometry/BezierCurve.hpp"
+#include "axom/primal/geometry/CurvedPolygon.hpp"
+#include "axom/primal/operators/detail/winding_number_impl.hpp"
+
+// C++ includes
+#include <cmath>
+
+namespace axom
+{
+namespace primal
+{
+/*
+ * \brief Compute the winding number with respect to a line segment
```

Comment:
There could conceivably be a function `winding_number( Segment<T, 3> )`, but it would always return 0. I'll definitely clarify things in the doxystring, but I also like the idea of separating the 2D and 3D functions with doxygen sections. Is it still possible to have the individual doxygen headers for each function inside the larger group?

Reply:
Yes -- the individual doxygen comments remain, but it adds an extra level of grouping.

Here's an example from `sidre::Group`: 

![image](https://github.com/LLNL/axom/assets/5626552/0b4b42b3-b858-413b-8217-a293f4d11b0d)
https://axom.readthedocs.io/en/develop/doxygen/html/classaxom_1_1sidre_1_1Group.html